dot/state/pruner/pruner.go

```diff
@@ -163,16 +167,16 @@ func (p *FullNode) addDeathRow(jr *journalRecord, blockNum int64) {
 		return
 	}
 
-	p.processInsertedKeys(jr.insertedKeys, jr.blockHash)
+	p.processInsertedKeys(jr.insertedHashesSet, jr.blockHash)
 
 	// add deleted keys from journal to death index
-	for _, k := range jr.deletedKeys {
+	for k := range jr.deletedHashesSet {
 		p.deathIndex[k] = blockNum
 	}
 
 	deletedKeys := make(map[common.Hash]int64)
-	for _, data := range jr.deletedKeys {
-		deletedKeys[data] = blockNum
+	for k := range jr.deletedHashesSet {
+		deletedKeys[k] = blockNum
```

Comment:
could we use the iteration from line 173 to populate the `deletedKeys`?

Reply:
Totally. Done in 3eac2917c0f9f44c5dfed109d63d0f35860de7d5